### ISSUE TITLE
Inconsistent validation of request payload in tenant registration operations.

### ISSUE THREAD

**Issue by kaniyan** [eclipse/hono#1555]
There has been some inconsistency in the validation of the request payload during tenant registration operations in the _file based device registry_. In some cases, it's been able to _add_ or _update_ a tenant even with attributes containing _invalid_ values.

For example, it's been **able** to _add_ or _update_ a tenant with invalid _minimum message size_ or  _sampling mode_. After that the _get_ tenant operation returns `400 Bad Request Error`.
 
```
{
    "enabled": true,
    "minimum-message-size": -100,
    "tracing":{
    	"sampling-mode": "xyz"
    }
}
```

On the other hand, it is **not able** to `add` or `update` a tenant with the below invalid request payload containing negative value for `max-connections`. In this case `400 Bad Request Error` is returned as expected.

```
{
    "enabled": true,
    "resource-limits": {
        "max-connections": -100
    }
}
```
**Findings:** 

In the _file based device registry_, while performing _add_ and _update_ operations, the request payload is validated and mapped to `TenantObject` and then stored in a file. And they are not validated against the Tenant Management API. A solution shall be, that before accepting the payload, it should also be validated against the `org.eclipse.hono.service.management.tenant.Tenant`.

**Comment by sophokles73** [comment]
@kaniyan can this be closed?

**Comment by kaniyan** [comment]
This has been closed through PR #1562.
